### PR TITLE
cudaPackages: bump default cudaPackages_11_7 -> cudaPackages_11_8

### PR DESCRIPTION
**Pull request by nviets**
###### Description of changes

Updated the default cudaPackages from 11.7 to 11.8 to address https://github.com/NixOS/nixpkgs/issues/222778. Pytorch doesn't support 12, yet, so this is an interim change. A nixpkgs-review with cudaSupport enabled is forthcoming.

Replaces 224927.

@ConnorBaker, @samuela, @SomeoneSerge - ugh, sorry about 224927. I think I managed to salvage the changes here.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin

### CODE REVIEW

**Comment by nviets** [comment]
Result of `nixpkgs-review pr 238622 --extra-nixpkgs-config '{ cudaCapabilities = [ "8.6" ]; }'` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>24 packages marked as broken and skipped:</summary>
  <ul>
    <li>cudaPackages.nvidia_driver</li>
    <li>python310Packages.dalle-mini</li>
    <li>python310Packages.dalle-mini.dist</li>
    <li>python310Packages.distrax</li>
    <li>python310Packages.distrax.dist</li>
    <li>python310Packages.elegy</li>
    <li>python310Packages.elegy.dist</li>
    <li>python310Packages.flax</li>
    <li>python310Packages.flax.dist</li>
    <li>python310Packages.mask-rcnn</li>
    <li>python310Packages.mask-rcnn.dist</li>
    <li>python310Packages.optuna</li>
    <li>python310Packages.optuna.dist</li>
    <li>python310Packages.rlax</li>
    <li>python310Packages.rlax.dist</li>
    <li>python310Packages.theanoWithCuda</li>
    <li>python310Packages.theanoWithCuda.dist</li>
    <li>python310Packages.treex</li>
    <li>python310Packages.treex.dist</li>
    <li>python310Packages.vqgan-jax</li>
    <li>python310Packages.vqgan-jax.dist</li>
    <li>python311Packages.theanoWithCuda</li>
    <li>python311Packages.theanoWithCuda.dist</li>
    <li>truecrack-cuda</li>
  </ul>
</details>
<details>
  <summary>18 packages failed to build:</summary>
  <ul>
    <li>cudaPackages.tensorrt (cudaPackages.tensorrt_8_5_3)</li>
    <li>cudaPackages.tensorrt.dev (cudaPackages.tensorrt_8_5_3.dev)</li>
    <li>cudaPackages.tensorrt_8_5_1</li>
    <li>cudaPackages.tensorrt_8_5_1.dev</li>
    <li>cudaPackages.tensorrt_8_5_2</li>
    <li>cudaPackages.tensorrt_8_5_2.dev</li>
    <li>cudaPackages.tensorrt_8_6_0</li>
    <li>cudaPackages.tensorrt_8_6_0.dev</li>
    <li>katagoTensorRT</li>
    <li>mathematica-cuda</li>
    <li>nvidia-thrust-intel</li>
    <li>python310Packages.scikit-tda</li>
    <li>python310Packages.scikit-tda.dist</li>
    <li>python310Packages.tensorrt</li>
    <li>python310Packages.tensorrt.dist</li>
    <li>python310Packages.tiny-cuda-nn</li>
    <li>python311Packages.tensorrt</li>
    <li>python311Packages.tensorrt.dist</li>
  </ul>
</details>
<details>
  <summary>193 packages built:</summary>
  <ul>
    <li>colmapWithCuda</li>
    <li>cudaPackages.cuda-samples</li>
    <li>cudaPackages.cuda_cccl</li>
    <li>cudaPackages.cuda_cudart</li>
    <li>cudaPackages.cuda_cuobjdump</li>
    <li>cudaPackages.cuda_cupti</li>
    <li>cudaPackages.cuda_cuxxfilt</li>
    <li>cudaPackages.cuda_demo_suite</li>
    <li>cudaPackages.cuda_documentation</li>
    <li>cudaPackages.cuda_gdb</li>
    <li>cudaPackages.cuda_memcheck</li>
    <li>cudaPackages.cuda_nsight</li>
    <li>cudaPackages.cuda_nvcc</li>
    <li>cudaPackages.cuda_nvdisasm</li>
    <li>cudaPackages.cuda_nvml_dev</li>
    <li>cudaPackages.cuda_nvprof</li>
    <li>cudaPackages.cuda_nvprune</li>
    <li>cudaPackages.cuda_nvrtc</li>
    <li>cudaPackages.cuda_nvtx</li>
    <li>cudaPackages.cuda_nvvp</li>
    <li>cudaPackages.cuda_profiler_api</li>
    <li>cudaPackages.cuda_sanitizer_api</li>
    <li>cudatoolkit (cudaPackages.cudatoolkit ,cudatoolkit_11)</li>
    <li>cudatoolkit.doc (cudaPackages.cudatoolkit.doc ,cudatoolkit_11.doc)</li>
    <li>cudatoolkit.lib (cudaPackages.cudatoolkit.lib ,cudatoolkit_11.lib)</li>
    <li>cudaPackages.cudnn (cudaPackages.cudnn_8_9)</li>
    <li>cudaPackages.cudnn_8_6</li>
    <li>cudaPackages.cudnn_8_7</li>
    <li>cudaPackages.cudnn_8_8</li>
    <li>cudaPackages.cutensor</li>
    <li>cudaPackages.cutensor.dev</li>
    <li>cudaPackages.fabricmanager</li>
    <li>cudaPackages.libcublas</li>
    <li>cudaPackages.libcufft</li>
    <li>cudaPackages.libcufile</li>
    <li>cudaPackages.libcurand</li>
    <li>cudaPackages.libcusolver</li>
    <li>cudaPackages.libcusparse</li>
    <li>cudaPackages.libnpp</li>
    <li>cudaPackages.libnvidia_nscq</li>
    <li>cudaPackages.libnvjpeg</li>
    <li>cudaPackages.nccl</li>
    <li>cudaPackages.nccl.dev</li>
    <li>cudaPackages.nsight_compute</li>
    <li>cudaPackages.nsight_systems</li>
    <li>cudaPackages.nvidia_fs</li>
    <li>faissWithCuda</li>
    <li>faissWithCuda.demos</li>
    <li>forge</li>
    <li>frigate</li>
    <li>gpt2tc</li>
    <li>gpu-burn</li>
    <li>gromacsCudaMpi</li>
    <li>gwe</li>
    <li>haskellPackages.tensorflow</li>
    <li>haskellPackages.tensorflow-core-ops</li>
    <li>haskellPackages.tensorflow-core-ops.doc</li>
    <li>haskellPackages.tensorflow-logging</li>
    <li>haskellPackages.tensorflow-logging.doc</li>
    <li>haskellPackages.tensorflow-ops</li>
    <li>haskellPackages.tensorflow-ops.doc</li>
    <li>haskellPackages.tensorflow.doc</li>
    <li>hip-nvidia</li>
    <li>hip-nvidia.doc</li>
    <li>katagoWithCuda</li>
    <li>librealsenseWithCuda</li>
    <li>librealsenseWithCuda.dev</li>
    <li>libtensorflow (libtensorflow.python)</li>
    <li>magma (magma-cuda ,magma_2_7_1)</li>
    <li>magma_2_6_2</li>
    <li>nvidia-thrust</li>
    <li>nvidia-thrust-cuda</li>
    <li>nvtop</li>
    <li>nvtop-nvidia</li>
    <li>openai-full</li>
    <li>openai-full.dist</li>
    <li>python310Packages.arviz</li>
    <li>python310Packages.arviz.dist</li>
    <li>python310Packages.bambi</li>
    <li>python310Packages.bambi.dist</li>
    <li>python310Packages.baselines</li>
    <li>python310Packages.baselines.dist</li>
    <li>python310Packages.bitsandbytes</li>
    <li>python310Packages.bitsandbytes.dist</li>
    <li>python310Packages.ctranslate2</li>
    <li>python310Packages.ctranslate2.dist</li>
    <li>python310Packages.cupy</li>
    <li>python310Packages.cupy.dist</li>
    <li>python310Packages.dm-sonnet</li>
    <li>python310Packages.dm-sonnet.dist</li>
    <li>python310Packages.edward</li>
    <li>python310Packages.edward.dist</li>
    <li>python310Packages.fastai</li>
    <li>python310Packages.fastai.dist</li>
    <li>python310Packages.faster-whisper</li>
    <li>python310Packages.faster-whisper.dist</li>
    <li>python310Packages.gpt-2-simple</li>
    <li>python310Packages.gpt-2-simple.dist</li>
    <li>python310Packages.jaxlibWithCuda</li>
    <li>python310Packages.jaxlibWithCuda.dist</li>
    <li>python310Packages.k-diffusion</li>
    <li>python310Packages.k-diffusion.dist</li>
    <li>python310Packages.langchain</li>
    <li>python310Packages.langchain.dist</li>
    <li>python310Packages.n3fit</li>
    <li>python310Packages.n3fit.dist</li>
    <li>python310Packages.numbaWithCuda</li>
    <li>python310Packages.numbaWithCuda.dist</li>
    <li>python310Packages.numpyro</li>
    <li>python310Packages.numpyro.dist</li>
    <li>python310Packages.openai-triton</li>
    <li>python310Packages.openai-triton-bin</li>
    <li>python310Packages.openai-triton-bin.dist</li>
    <li>python310Packages.openai-triton.dist</li>
    <li>python310Packages.pot</li>
    <li>python310Packages.pot.dist</li>
    <li>python310Packages.pycuda</li>
    <li>python310Packages.pycuda.dist</li>
    <li>python310Packages.pymc</li>
    <li>python310Packages.pymc.dist</li>
    <li>python310Packages.pynvml</li>
    <li>python310Packages.pynvml.dist</li>
    <li>python310Packages.pyrealsense2WithCuda</li>
    <li>python310Packages.pyrealsense2WithCuda.dev</li>
    <li>python310Packages.skrl</li>
    <li>python310Packages.skrl.dist</li>
    <li>python310Packages.spacy</li>
    <li>python310Packages.spacy-loggers</li>
    <li>python310Packages.spacy-loggers.dist</li>
    <li>python310Packages.spacy-transformers</li>
    <li>python310Packages.spacy-transformers.dist</li>
    <li>python310Packages.spacy.dist</li>
    <li>python310Packages.tensorflow (python310Packages.tensorflow-build ,python310Packages.tensorflowWithoutCuda)</li>
    <li>python310Packages.tensorflow.dist (python310Packages.tensorflow-build.dist ,python310Packages.tensorflowWithoutCuda.dist)</li>
    <li>python310Packages.tensorflow-datasets</li>
    <li>python310Packages.tensorflow-datasets.dist</li>
    <li>python310Packages.tensorflow-probability</li>
    <li>python310Packages.tensorflow-probability.dist</li>
    <li>python310Packages.tensorflowWithCuda</li>
    <li>python310Packages.tensorflowWithCuda.dist</li>
    <li>python310Packages.textacy</li>
    <li>python310Packages.textacy.dist</li>
    <li>python310Packages.textnets</li>
    <li>python310Packages.textnets.dist</li>
    <li>python310Packages.tflearn</li>
    <li>python310Packages.tflearn.dist</li>
    <li>python310Packages.torchWithCuda</li>
    <li>python310Packages.torchWithCuda.dev</li>
    <li>python310Packages.torchWithCuda.dist</li>
    <li>python310Packages.torchWithCuda.lib</li>
    <li>python310Packages.torchWithRocm</li>
    <li>python310Packages.torchWithRocm.dev</li>
    <li>python310Packages.torchWithRocm.dist</li>
    <li>python310Packages.torchWithRocm.lib</li>
    <li>python310Packages.torchaudio-bin</li>
    <li>python310Packages.torchaudio-bin.dist</li>
    <li>python310Packages.torchvision-bin</li>
    <li>python310Packages.torchvision-bin.dist</li>
    <li>python310Packages.trainer</li>
    <li>python310Packages.trainer.dist</li>
    <li>python310Packages.trfl</li>
    <li>python310Packages.trfl.dist</li>
    <li>python310Packages.umap-learn</li>
    <li>python310Packages.umap-learn.dist</li>
    <li>python310Packages.wandb</li>
    <li>python310Packages.wandb.dist</li>
    <li>python311Packages.cupy</li>
    <li>python311Packages.cupy.dist</li>
    <li>python311Packages.jaxlibWithCuda</li>
    <li>python311Packages.jaxlibWithCuda.dist</li>
    <li>python311Packages.openai-triton</li>
    <li>python311Packages.openai-triton-bin</li>
    <li>python311Packages.openai-triton-bin.dist</li>
    <li>python311Packages.openai-triton.dist</li>
    <li>python311Packages.pycuda</li>
    <li>python311Packages.pycuda.dist</li>
    <li>python311Packages.pynvml</li>
    <li>python311Packages.pynvml.dist</li>
    <li>python311Packages.pyrealsense2WithCuda</li>
    <li>python311Packages.pyrealsense2WithCuda.dev</li>
    <li>python311Packages.torchaudio-bin</li>
    <li>python311Packages.torchaudio-bin.dist</li>
    <li>python311Packages.torchvision-bin</li>
    <li>python311Packages.torchvision-bin.dist</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.tensorflow</li>
    <li>tiny-cuda-nn</li>
    <li>tts</li>
    <li>tts.dist</li>
    <li>wyoming-faster-whisper</li>
    <li>wyoming-faster-whisper.dist</li>
    <li>xgboostWithCuda</li>
    <li>xpraWithNvenc</li>
    <li>xpraWithNvenc.dist</li>
  </ul>
</details>

**Comment by nviets** [comment]
###### Failed derivations

- [ ] Mathematica_13.2.1_BNDL_LINUX.sh.drv (derivation hash: r4z546rdvaswc0112l0faf6m4g1lyzbw)
  - https://gist.github.com/nviets/1b04bc45bb352bc070023d324500c3a7
- [ ] TensorRT-8.5.1.7.Linux.x86_64-gnu.cuda-11.8.cudnn8.6.tar.gz.drv (derivation hash: 4zgw9b5g7zqxv7bdgg0a2d0ilb41cic4)
  - https://gist.github.com/nviets/a165dfd4d120db6af6a6916ba3b3b49d
- [ ] TensorRT-8.5.2.2.Linux.x86_64-gnu.cuda-11.8.cudnn8.6.tar.gz.drv (derivation hash: jzhfmpv57mjv76snk69hlg4zmbhnz4mz)
  - https://gist.github.com/nviets/043520193e9b53b6a700ee1bf9650cbb
- [ ] TensorRT-8.5.3.1.Linux.x86_64-gnu.cuda-11.8.cudnn8.6.tar.gz.drv (derivation hash: 2d73hhfzzpf0dqmwmhymdni0g7pfrnb8)
  - https://gist.github.com/nviets/c48a28b5b725359a3e27c03f208a33ab
- [ ] TensorRT-8.6.0.12.Linux.x86_64-gnu.cuda-11.8.tar.gz.drv (derivation hash: zkdxf2s26a5vzp1a1jiwc0px1rdch06b)
  - https://gist.github.com/nviets/aabef0d3e99a56060b72ddbf99a123db
- [ ] nvidia-thrust-1.16.0.drv (derivation hash: dqamjckc3vn6birbnsscpgfw0ilfqngc)
  - https://gist.github.com/nviets/3ca651f46dba2b0f946e8916a1e4b5e1
- [ ] python3.10-kmapper-2.0.1.drv (derivation hash: l0l51khayqj1aqb8hy0if7689j3gv2y8)
  - https://gist.github.com/nviets/72d0af22fc8ba4184fb4dbf52bb1545e
- [ ] tiny-cuda-nn.drv (derivation hash: srz60hla8ndwv5rm6r9i5wf3r25nk665)
  - https://gist.github.com/nviets/826131ec117ea6e18f73f4c88fb50d4a

**Comment by nviets** [comment]
tiny-cuda-nn is hopefully fixed by #238619.
kmapper is to be flagged as broken #235004
nvidia-thrust fails to build, but there's an open PR to upgrade it: #224868

@samuela, @SomeoneSerge - can we call this ready for a merge?

File: pkgs/test/cuda/cuda-samples/generic.nix [hunk 1]
```diff
@@ -1,14 +1,18 @@
 { lib
-, stdenv
+, cudaPackages
+, gcc11Stdenv
```
Comment (samuela) [inline pkgs/test/cuda/cuda-samples/generic.nix:3]: where is this used?

File: pkgs/test/cuda/cuda-samples/generic.nix [hunk 2]
```diff
@@ -1,14 +1,18 @@
...
 , pkg-config
 , sha256
+, glfw3
+, freeimage
 }:
-
-stdenv.mkDerivation rec {
+let
+  stdenv = cudaPackages.backendStdenv;
+in stdenv.mkDerivation rec {
```
Comment (samuela) [inline pkgs/test/cuda/cuda-samples/generic.nix:15]:
```suggestion
cudaPackages.backendStdenv.mkDerivation rec {
```

File: pkgs/test/cuda/cuda-samples/generic.nix (hunk 1 above)
Comment (nviets) [reply pkgs/test/cuda/cuda-samples/generic.nix:3]: removing it.

**Comment by samuela** [comment]
Thanks @nviets !
